diff --git a/dist/container/index.js b/dist/container/index.js
index be50f413df7f8696d5d7ce95c441c3e339f11b7d..6c27f1e0f5002a1f24154618aac2ad4c980d66f9 100644
--- a/dist/container/index.js
+++ b/dist/container/index.js
@@ -21,7 +21,7 @@ function createManifest(manifest, renderers, middleware) {
     };
   }
   return {
-    hrefRoot: import.meta.url,
+    hrefRoot: (() => { try { new URL(import.meta.url); return import.meta.url; } catch { return "file:///app/"; } })(),
     srcDir: manifest?.srcDir ?? ASTRO_CONFIG_DEFAULTS.srcDir,
     buildClientDir: manifest?.buildClientDir ?? ASTRO_CONFIG_DEFAULTS.build.client,
     buildServerDir: manifest?.buildServerDir ?? ASTRO_CONFIG_DEFAULTS.build.server,
